trojsten/notifications/forms.py

```diff
@@ -0,0 +1,65 @@
+from crispy_forms import layout
+from crispy_forms.helper import FormHelper
+from django import forms
+from django.utils.translation import ugettext_lazy as _
+from django_nyt.models import Subscription
+
+from trojsten.notifications import constants
+from trojsten.notifications.models import IgnoredNotifications
+from trojsten.notifications.utils import unsubscribe_from_subscription
+
+
+def _pretty_subscription_name(subscription):
+    pretty_name = constants.NOTIFICATION_PRETTY_NAMES[subscription.notification_type.key]
+    if subscription.object_id:
+        target = (
+            subscription.notification_type.content_type.model_class()
+            .objects.filter(pk=subscription.object_id)
+            .get()
+        )
+        return "%s (%s)" % (pretty_name, target)
+    else:
+        return "%s" % (pretty_name,)
+
+
+class NotificationSettingsForm(forms.Form):
+    def __init__(self, user, *args, **kwargs):
+        super(NotificationSettingsForm, self).__init__(*args, **kwargs)
+        self.user = user
+
+        self.helper = FormHelper()
+        self.helper.add_input(layout.Submit("notification_subscription_submit", _("Submit")))
+        self.helper.form_show_labels = True
+
+        subscriptions = Subscription.objects.filter(settings__user=self.user).all()
+        self.fields["subscriptions"] = forms.MultipleChoiceField(
+            choices=[(s.pk, _pretty_subscription_name(s)) for s in subscriptions],
+            initial=[s.pk for s in subscriptions],
+            widget=forms.CheckboxSelectMultiple,
+            required=False,
+        )
+        self.fields["subscriptions"].label = _("Subscribed events")
+
+        ignored = IgnoredNotifications.objects.filter(user=self.user).all()
```

Comment:
Podla mna nepotrebujeme zvlast form na ignored notifications. To je len implementacny detajl a pre uzivatela to moze byt matuce. Spravne spravanie by malo byt automaticky sa subscribne pri submite a ked sa clovek sam unsubscribne, tak sa to prida do blacklistu (vzdy) aby sa uz automaticky nesubscribol. Potom sa clovek moze subscribnut manualne.

Reply:
No, ono aktuálne nie je úplne spôsob sa manuálne subscribnúť. Neviem, kde by sme toto nastavenie chceli, ale napr. by to mohlo byť na stránke pri úlohách/kolách by si mal tlačítko "Subscribe". (možno aj unsubscribe?)
Potom by samozrejme ignored mohlo ísť preč.

Lebo neviem úplne ako by to šlo ináč. Keby sme to dali medzi nastavenia, aktuálne nevieš, čo všetko sa dá subscribovať a aké key/target-y sú k dispozícií.

Reply:
Čo sa stane ak odškrtneš nejaký subscription?

Podľa mňa by bolo fajn mať v tých nastaveniach checkbox pre každý možný subscription a ak je zaskrtnuty subscribujes, inak nie.